### ISSUE TITLE
Menu-check icon is too big in current theme

### ISSUE THREAD

**Issue by dryeo** [bitwiseworks/mozilla-os2#252]
With current 45ESR, the make-check.gif is displayed as too big. This affects Firefox, Thunderbird and especially SeaMonkey which uses the icon more.
See picture.
file:///H:/tmp/menu-check.45.png
Using the toolkit/themes/os2/global/menu.css instead of the current one fixes this,
file:///H:/tmp/menu-check-good.png

Easiest way to test is to apply this,
```
diff --git a/toolkit/themes/os2/global/popup.css b/toolkit/themes/os2/global/popup.css
index 250e63e7d8..a6603fa529 100644
--- a/toolkit/themes/os2/global/popup.css
+++ b/toolkit/themes/os2/global/popup.css
@@ -21,16 +21,15 @@ panel {
   color:MenuText;
 }
 
-menupopup > menu > menupopup {
-  /* align submenus in popups */
-  -moz-margin-start: 2px;
-  margin-top: 1px;
+menupopup {
+  border-top: 1px solid;
+  -moz-border-top-colors: ThreeDHighlight;
 }
 
-menubar > menu > menupopup {
-  /* align submenus in menubar */
-  -moz-margin-start: -4px;
-  margin-top: -1px;
+menupopup > menu > menupopup {
+ /* align submenus */
+  -moz-margin-start: 2px;
+  margin-top: 2px;
 }
 
 panel[type="arrow"] {
```
to \dist\bin\chrome\toolkit\skin\classic\global>e\toolkit\skin\classic\global\menu.css and run \dist\bin\firefox.exe

**Comment by dryeo** [comment]
Hmm, screenshots won't upload. You can see the large menu-check gif by going to View-->Page Style and looking at the left of the Basic Page Style menu item in Firefox.

**Comment by dmik** [comment]
@dryeo hmm, your patch doesn't seem to help here. FIrst, it is for `popup.css` but you say that one should update `menu.css`. Second, applying your patch to `popup.css` doesn't make the checkbox look smaller, it still looks the same here, like this (both with or without your patch):

![2018-04-13 15 52 25](https://user-images.githubusercontent.com/900726/38735827-b5dae420-3f32-11e8-919b-c01890a420b4.png)

(BTW, in order to properly attach images here, you should place the "selecting them" link under the comment entry field, I'm not sure local file DND works properly on OS/2 here).

**Comment by dmik** [comment]
I can't figure out a proper solution for that. Something has changed in Firefox wrt `list-style-image` processing so the intrinsic image width and height are ignored now (or the fact that it's a raster image).

**Comment by dryeo** [comment]
I'll send a patch after Firefox finishes building here.

**Comment by dryeo** [comment]
Sorry for sending the wrong patch earlier, this seems to work. Note that the root cause seems to be a bad merge and this is based on an earlier release.
```
diff --git a/toolkit/themes/os2/global/menu.css b/toolkit/themes/os2/global/menu.css
index 050f76c16d..061bc95089 100644
--- a/toolkit/themes/os2/global/menu.css
+++ b/toolkit/themes/os2/global/menu.css
@@ -12,7 +12,6 @@
 
 menu,
 menuitem,
-menucaption,
 .splitmenu-menuitem {
   -moz-appearance: menuitem;
   -moz-box-align: center;
@@ -24,6 +23,10 @@ menucaption,
   -moz-image-region: auto;
 }
 
+menu:not(:-moz-lwtheme) {
+  background-color: Menu;
+}
+
 menuitem[default="true"] {
   font-weight: bold;
 }
@@ -32,10 +35,9 @@ menu[disabled="true"],
 menuitem[disabled="true"],
 .splitmenu-menuitem[disabled="true"],
 menu[_moz-menuactive="true"][disabled="true"],
-menuitem[_moz-menuactive="true"][disabled="true"],
+menuitem[_moz-menuactive="true"][disabled="true"]
 .splitmenu-menuitem[_moz-menuactive="true"][disabled="true"] {
   color: GrayText;
-  text-shadow: none;
 }
 
 menuitem.spell-suggestion {
@@ -48,27 +50,16 @@ menuitem.spell-suggestion {
 .menu-iconic-accel,
 .menu-text,
 .menu-iconic-text {
-  margin: 0px !important;
-  padding: 1px 0px;
+  margin-top: 0 !important;
+  margin-bottom: 0 !important;
+  -moz-margin-start: 0px !important;
+  -moz-margin-end: 0px !important;
   color: inherit;
 }
 
 .menu-text {
-  -moz-padding-start: 1.45em !important;
-  -moz-appearance: menuitemtext;
-}
-
-.menu-text,
-.menu-iconic-text {
+  -moz-margin-start: 21px !important;
   font-weight: inherit;
-  -moz-margin-start: 2px !important;
-  -moz-padding-end: 2px;
-}
-
-menucaption > .menu-text,
-menucaption > .menu-iconic-text {
-  font-weight: bold;
-  -moz-padding-start: 0 !important;
 }
 
 .menu-description {
@@ -79,20 +70,24 @@ menucaption > .menu-iconic-text {
 
 .menu-accel,
 .menu-iconic-accel {
+  -moz-margin-end: 19px !important;
+  -moz-margin-start: 8px !important;
   color: inherit;
-  -moz-margin-start: 0.74em !important;
-  -moz-margin-end: 1.35em !important;
+}
+
+.menu-iconic-text {
+  -moz-margin-start: 3px !important;
 }
 
 .menu-iconic-left {
-  min-width: 1.45em;
+  -moz-margin-start: 2px !important;
+  min-width: 16px;
+  min-height: 16px;
 }
 
 .menu-iconic-icon {
-  width: 16px;
-  height: 16px;
-  margin-top: -1px !important;
-  margin-bottom: 1px !important;
+  max-width: 16px;
+  max-height: 16px;
 }
 
 menu.menu-iconic > .menu-iconic-left,
@@ -105,7 +100,6 @@ menuitem.menuitem-iconic > .menu-iconic-left,
 /* ..... menu arrow box ..... */
 
 .menu-right {
-  -moz-appearance: menuarrow;
   margin-top: 2px;
   margin-bottom: 3px;
   -moz-padding-end: 6px;
@@ -138,8 +132,7 @@ menubar > menu:-moz-lwtheme:not([disabled="true"]) {
   color: inherit !important;
 }
 
-menubar > menu:-moz-lwtheme[_moz-menuactive="true"]:not([disabled="true"]) {
-  background-color: Highlight;
+menubar > menu:-moz-lwtheme[_moz-menuactive="true"] {
   color: HighlightText !important;
   text-shadow: none;
 }
@@ -157,65 +150,44 @@ menubar > menu:-moz-lwtheme[_moz-menuactive="true"]:not([disabled="true"]) {
 /* ::::: menu/menuitems in popups ::::: */
 
 menupopup > menu,
-menupopup > menuitem,
-menupopup > menucaption {
+menupopup > menuitem {
   max-width: 42em;
+  padding-bottom: 2px;
 }
 
 menu[_moz-menuactive="true"],
 menuitem[_moz-menuactive="true"],
 .splitmenu-menuitem[_moz-menuactive="true"] {
-  background-color: -moz-menuhover;
-  color: -moz-menuhovertext;
+  background: -moz-MenuHover;
+  color: -moz-MenuHoverText;
 }
 
 /* ::::: menu/menuitems in menulist popups ::::: */
 
+.menulist-menupopup > menuitem,
 menulist > menupopup > menuitem,
-menulist > menupopup > menucaption,
+.menulist-menupopup > menu,
 menulist > menupopup > menu {
-  -moz-appearance: none !important;
   border: 1px solid transparent;
-  -moz-padding-start: 5px;
-  -moz-padding-end: 5px;
+  padding: 1px 5px;
   max-width: none;
   font: message-box;
   color: -moz-FieldText;
+  background-color: -moz-Field;
 }
 
-menulist > menupopup > menuitem[_moz-menuactive="true"],
-menulist > menupopup > menu[_moz-menuactive="true"] {
-  background-color: highlight;
-  color: highlighttext;
-}
-
+.menulist-menupopup > menuitem > .menu-iconic-left,
 menulist > menupopup > menuitem > .menu-iconic-left,
-menulist > menupopup > menucaption > .menu-iconic-left,
+.menulist-menupopup > menu > .menu-iconic-left,
 menulist > menupopup > menu > .menu-iconic-left {
   display: none;
 }
 
-menulist > menupopup > menuitem > label,
-menulist > menupopup > menucaption > label,
-menulist > menupopup > menu > label {
-  padding-top: 0px;
-  padding-bottom: 0px;
-}
-
-menulist:-moz-focusring > menupopup > menuitem[_moz-menuactive="true"] {
+menulist > menupopup > menuitem[_moz-menuactive="true"] {
   border: 1px dotted #F5DB95;
 }
 
-menulist > menupopup > menuitem[_moz-menuactive="true"][disabled="true"] {
-  color: GrayText;
-}
-
-menulist > menupopup > menuitem[disabled="true"]:not([_moz-menuactive="true"]):-moz-system-metric(windows-classic) {
-  color: GrayText;
-  text-shadow: none;
-}
-
-menulist > menupopup > :-moz-any(menuitem, menucaption):not(.menuitem-iconic) > .menu-iconic-text {
+menulist > menupopup > menuitem > .menu-iconic-text {
   margin: 0 !important;
 }
 
@@ -226,11 +198,11 @@ menuitem[checked="true"] {
   list-style-image: url("chrome://global/skin/menu/menu-check.gif");
   -moz-image-region: auto;
 }
+/* menuitem[checked="true"][type="radio"] */
 
 /* ::::: menuseparator ::::: */
 
 menuseparator {
-  -moz-appearance: menuseparator;
   margin: 3px 0 3px 0 !important;
   border-top: 1px solid ThreeDShadow;
   border-bottom: 1px solid ThreeDHighlight;
@@ -257,3 +229,4 @@ menulist > menupopup > menuseparator,
   list-style-image: url("chrome://global/skin/columnselect.gif");
   -moz-image-region: auto;
 }
+
```

**Comment by dryeo** [comment]
Selecting the image doesn't seem to work here either.

**Comment by dmik** [comment]
Ok, in your patch you basically reverted 05f6e7d3551413495c1b7d47b1d8bdc8285d6004 where I fixed a lot of glitches (making menus look less native) that appeared in esr45. Not good :) I found that here was a just a couple of properties missing necessary to make it work as before. Fixed in c0a80586e9. Thanks for reporting and for regression testing.
